### PR TITLE
Application Skeleton Implementation

### PR DESCRIPTION
**Pull request by Ishimwe7**
**Description**
This PR introduces skeleton loading components throughout the application to improve user experience by providing visual feedback during data fetching. These skeleton loaders mimic the structure of the final content, making the interface more intuitive and responsive while the content is loading.

**Description of Task Completed**

- Implemented skeleton loading components across key areas of the application.
- Skeleton loaders are designed to represent the final structure of the content, enhancing the visual feedback to users during loading times.
- Integrated loaders with existing features to ensure a smooth transition when content is being fetched from the backend.

**How to test this PR(locally)**

- Clone the repo and check out the branch ft-skeleton-implementation
- Set up environment variables as indicated in .env.example
- Run npm install followed by npm run dev
- Test loading states in following areas:
     - Navigate through different sections
     - Observe skeleton animations during data fetching
     - Verify smooth transitions between states

**Related Issue**
_https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/161_ 

**Screenshots** 

NavbarMembers

traineeApplicant

editApplicantInfo

trash

### CODE REVIEW

**Review by Ismaelmurekezi — APPROVED** [review]

**Review by SaddockAime — APPROVED** [review]

**Review by mugishaj092 — APPROVED** [review]

File: src/pages/ApplicationCycle/ApplicationCycle.tsx [hunk 1]
```diff
@@ -16,15 +16,23 @@ import * as BsIcons from "react-icons/bs";
...
 const ApplicationCycle = (props: any) => {
   const { allCycles, errors } = props;
-
+  const [loading, setLoading] = useState(true);
   const cycles = allCycles.data;
 
+  // useEffect(() => {
+  //   props.getAllCycles();
+  // }, []);
+
```
Comment (ncutixavier) [inline src/pages/ApplicationCycle/ApplicationCycle.tsx]: Get rid of all unnecessary comments

File: src/pages/TrainneeDetails.tsx [hunk 2]
```diff
@@ -20,6 +21,7 @@ const TrainneeDetails = (props: any) => {
   }, [ID]);
 
   const traineeDetails = oneTraineeDetails.data;
+  console.log("Trainee: ", traineeDetails);
```
Comment (ncutixavier) [inline src/pages/TrainneeDetails.tsx]: remove all logs

Comment (Ishimwe7) [reply src/pages/TrainneeDetails.tsx]: let me remove them

**Comment by ncutixavier** [comment]
This PR has conflicts

**Comment by Ishimwe7** [comment]
> This PR has conflicts

Let me resolve them

**Comment by ncutixavier** [comment]
Your PR description is incompleted

**Comment by Ishimwe7** [comment]
> Your PR description is incompleted

I updated it.

**Pull request closed without merging**